src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +16,33 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = factory.openSession();
```

Comment:
The variable `factory` is used here, but it is not declared in this class. It seems like it should be `sessionFactory`, which is inherited from the `AbstractDao` class. Ensure that `sessionFactory` is correctly used to open sessions.